ocaml/snarky_js_constants.ml

```diff
@@ -18,6 +23,79 @@ let prefixes =
     ; ("deriveTokenId", `String (derive_token_id :> string))
     ]
 
+type hash_prefix_kind = Kimchi | Legacy
+
+let prefix_hash_entry (kind : hash_prefix_kind) (s : string) =
+  let s, fields =
+    match kind with
+    | Kimchi ->
+        (s, Random_oracle.(State.to_array (salt s)))
+    | Legacy ->
+        (s, Random_oracle.Legacy.(State.to_array (salt s)))
+  in
+  ((s :> string), array field fields)
+
+let prefix_hashes =
+  let open Hash_prefixes in
+  `Assoc
+    (List.map ~f:(prefix_hash_entry Kimchi)
+       [ (receipt_chain_user_command :> string)
+       ; (receipt_chain_zkapp :> string)
```

Comment:
these changes are about computing some hashes at compile time and put them in the `constants.ts` file